eZ/Publish/Core/Persistence/Legacy/Content/Type/Handler.php

```diff
@@ -401,8 +401,17 @@ public function copy($userId, $contentTypeId, $status)
         $createStruct->modifierId = $userId;
         $createStruct->created = $createStruct->modified = time();
         $createStruct->creatorId = $userId;
-        $createStruct->remoteId = substr(sha1(uniqid(get_class($createStruct), true)), 0, 7);
-        $createStruct->identifier = 'cpy_' . $createStruct->identifier . '_' . $createStruct->remoteId;
+        $originalRemoteId = $createStruct->remoteId;
+        // truncate remoteId to 32 chars to keep BC length of that field
+        $createStruct->remoteId = substr(sha1(uniqid(get_class($createStruct), true)), 0, 32);
+        $abbrLen = 7;
+        // prepare identifier of a copy as
+        // cp_<sourceIdentifier>_<abbreviatedSourceRemoteId>_<newAbbreviatedRemoteId>
+        // to avoid making it too long
+        $createStruct->identifier = 'cp_' .
```

Comment:
Could you move that to at least a private method ? It would allow you to give it a meaningful name, and add a phpdoc description, instead of having a long inline one.